Enable "nilnil" linter in golangci-lint

Enable "nilnil" linter in golangci-lint. Fix any warnings enabling it causes.

/area ci
/help
/triage accepted

this one a bit tricky, it's hard to fix it without project experience :( 

I only can enable linter and add `//nolint:nilnil` or so on those lines
```
INFO [runner] linters took 7.612566458s with stages: goanalysis_metalinter: 7.597061583s
controllers/metal3.io/baremetalhost_controller.go:713:3: return both a `nil` error and an invalid value: use a sentinel error instead (nilnil)
                return nil, nil
                ^
controllers/metal3.io/baremetalhost_controller.go:2040:4: return both a `nil` error and an invalid value: use a sentinel error instead (nilnil)
                        return nil, nil
                        ^
controllers/metal3.io/baremetalhost_controller.go:2143:3: return both a `nil` error and an invalid value: use a sentinel error instead (nilnil)
                return nil, nil
                ^
controllers/metal3.io/baremetalhost_controller.go:2156:3: return both a `nil` error and an invalid value: use a sentinel error instead (nilnil)
                return nil, nil
                ^
pkg/provisioner/ironic/ironic.go:283:2: return both a `nil` error and an invalid value: use a sentinel error instead (nilnil)
        return nil, nil
```

--- a/.gitignore
+++ b/.gitignore
@@ -60,3 +60,6 @@ config/overlays/temp
 .venv
 
 .idea/
+
+# VS Code settings
+.vscode/

--- a/.golangci.yaml
+++ b/.golangci.yaml
@@ -52,7 +52,7 @@ linters:
   - nakedret
   - nilerr
   - nilnesserr
-  #- nilnil
+  - nilnil
   - noctx
   - nolintlint
   - nosprintfhostport

--- a/internal/controller/metal3.io/baremetalhost_controller.go
+++ b/internal/controller/metal3.io/baremetalhost_controller.go
@@ -709,7 +709,7 @@ func getHostArchitecture(host *metal3api.BareMetalHost) string {
 func (r *BareMetalHostReconciler) getPreprovImage(info *reconcileInfo, formats []metal3api.ImageFormat) (*provisioner.PreprovisioningImage, error) {
 	if formats == nil {
 		// No image build requested
-		return nil, nil
+		return nil, nil //nolint:nilnil
 	}
 
 	if len(formats) == 0 {
@@ -2047,7 +2047,7 @@ func (r *BareMetalHostReconciler) acquireHostUpdatePolicy(info *reconcileInfo) (
 			// garbage collected. For additional cleanup logic use
 			// finalizers.  Return and don't requeue
 
-			return nil, nil
+			return nil, nil //nolint:nilnil
 		}
 		// Error reading the object
 		return nil, fmt.Errorf("could not load hostUpdatePolicy resource due to %w", err)
@@ -2150,7 +2150,7 @@ func (r *BareMetalHostReconciler) getHostStatusFromAnnotation(host *metal3api.Ba
 	annotations := host.GetAnnotations()
 	content := []byte(annotations[metal3api.StatusAnnotation])
 	if annotations[metal3api.StatusAnnotation] == "" {
-		return nil, nil
+		return nil, errors.New("status annotation not found")
 	}
 	objStatus, err := unmarshalStatusAnnotation(content)
 	if err != nil {
@@ -2163,7 +2163,7 @@ func (r *BareMetalHostReconciler) getHostStatusFromAnnotation(host *metal3api.Ba
 func (r *BareMetalHostReconciler) getHardwareDetailsFromAnnotation(host *metal3api.BareMetalHost) (*metal3api.HardwareDetails, error) {
 	annotations := host.GetAnnotations()
 	if annotations[metal3api.HardwareDetailsAnnotation] == "" {
-		return nil, nil
+		return nil, nil //nolint:nilnil
 	}
 	objHardwareDetails := &metal3api.HardwareDetails{}
 	decoder := json.NewDecoder(strings.NewReader(annotations[metal3api.HardwareDetailsAnnotation]))

--- a/pkg/provisioner/ironic/ironic.go
+++ b/pkg/provisioner/ironic/ironic.go
@@ -232,6 +232,7 @@ func (p *ironicProvisioner) isAddressAllocatedToPort(address string) (bool, erro
 }
 
 // Look for an existing registration for the host in Ironic.
+// Returns nil, nil if no host exists with MAC.
 func (p *ironicProvisioner) findExistingHost(bootMACAddress string) (ironicNode *nodes.Node, err error) {
 	// Try to load the node by UUID
 	ironicNode, err = p.getNode()
@@ -266,7 +267,7 @@ func (p *ironicProvisioner) findExistingHost(bootMACAddress string) (ironicNode 
 
 	if err != nil {
 		p.log.Info("failed to find an existing port with address", "MAC", bootMACAddress)
-		return nil, nil //nolint:nilerr
+		return nil, nil //nolint:nilerr,nilnil
 	}
 
 	if len(allPorts) > 0 {
@@ -291,7 +292,7 @@ func (p *ironicProvisioner) findExistingHost(bootMACAddress string) (ironicNode 
 	p.log.Info("port with address doesn't exist", "MAC", bootMACAddress)
 	// Either the node was never created or the Ironic database has
 	// been dropped.
-	return nil, nil
+	return nil, nil //nolint:nilnil
 }
 
 func (p *ironicProvisioner) createPXEEnabledNodePort(uuid, macAddress string) error {

